datafusion/core/src/physical_plan/windows/bounded_window_agg_exec.rs

```diff
@@ -428,10 +434,19 @@ impl SortedPartitionByBoundedWindowStream {
         input: SendableRecordBatchStream,
         baseline_metrics: BaselineMetrics,
         partition_by_sort_keys: Vec<PhysicalSortExpr>,
-    ) -> Self {
+        ordered_partition_by_indices: Vec<usize>,
+    ) -> Result<Self> {
         let state = window_expr.iter().map(|_| IndexMap::new()).collect();
         let empty_batch = RecordBatch::new_empty(schema.clone());
-        Self {
+
+        // In BoundedWindowAggExec all partition by columns should be ordered.
+        if window_expr[0].partition_by().len() != ordered_partition_by_indices.len() {
+            return Err(DataFusionError::Execution(
```

Comment:
```suggestion
            return Err(DataFusionError::Internal(
```

Reply:
This should only happen if there is a bug, right? The error won't happen on poorly formed sql, is that right?